kratos/python_scripts/eigen_solver_factory.py

```diff
@@ -1,30 +1,43 @@
 from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
 
-def ConstructSolver(settings):
-    import KratosMultiphysics
-    from KratosMultiphysics import python_linear_solver_factory as linear_solver_factory
+# Importing the Kratos Library
+import KratosMultiphysics as KM
+
+from KratosMultiphysics import python_linear_solver_factory as linear_solver_factory
+import KratosMultiphysics.kratos_utilities as kratos_utils
+
+if kratos_utils.CheckIfApplicationsAvailable("EigenSolversApplication"):
+    import KratosMultiphysics.EigenSolversApplication as EiSA
+
+if kratos_utils.CheckIfApplicationsAvailable("ExternalSolversApplication"):
+    import KratosMultiphysics.ExternalSolversApplication as ExSA
```

Comment:
same as above

Reply:
I trust you, I am a little lost with the imports

Reply:
Basic guide:
- Import the things you need for sure at the top of the file
- Do not import things you might not need (e.g. the solver-apps) => those will be imported by the corresponding factories